src/s3/util/progressReporter.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger'
+
+/**
+ * Returns a function that pipes cumulative byte progress to VSCode's incremental percentage Progress.
+ *
+ * @param progress the VSCode progress to update.
+ * @param totalBytes the total bytes.
+ * Used in combination with cumulative bytes to calculate incremental progress.
+ */
+export function progressReporter(
+    progress: vscode.Progress<{ message?: string; increment?: number }>,
+    totalBytes?: number
+): (loadedBytes: number) => void {
+    let report: ProgressReport = new ProgressReport(totalBytes)
+
+    return loadedBytes => {
+        report.updateLoadedBytes(loadedBytes)
+        if (report.incrementalPercentage) {
+            progress.report({ increment: report.incrementalPercentage })
+        }
+
+        getLogger().info(`${report}`)
```

Comment:
Is this a descriptive log? (I am unsure what output looks like for this)
Consider making this verbose.